src/kvstore/wal/FileBasedWal.cpp

```diff
@@ -672,23 +672,30 @@ void FileBasedWal::cleanWAL() {
 
 void FileBasedWal::cleanWAL(LogID id) {
   std::lock_guard<std::mutex> g(walFilesMutex_);
+  auto now = time::WallClock::fastNowInSec();
+  size_t index = 0;
   if (walFiles_.empty()) {
     return;
   }
+  auto size = walFiles_.size();
+  if (size < 2) {
+    return;
+  }
 
   if (walFiles_.rbegin()->second->lastId() < id) {
     VLOG(3) << "Try to clean wal not existed " << id << ", lastWal is "
             << walFiles_.rbegin()->second->lastId();
     return;
   }
-
+  int walTTL = FLAGS_wal_ttl;
   // remove wal file that lastId is less than target
   auto iter = walFiles_.begin();
   while (iter != walFiles_.end()) {
-    if (iter->second->lastId() < id) {
+    if (iter->second->lastId() < id && index < size - 2 && (now - iter->second->mtime() > walTTL)) {
```

Comment:
If we do not use walTTL to limit the cleanWAL, what will happen?

Reply:
Then no wal would be deleted